client/blocks/comment-detail/docs/comment-faker.jsx

```diff
@@ -28,59 +28,101 @@ export const CommentFaker = WrappedCommentList => class extends Component {
-		this.setState( {
-			comments: {
-				...this.state.comments,
-				[ commentId ]: {
-					...comment,
-					i_like: likeValue,
-					status: likeValue ? 'approved' : comment.status,
-				}
-			},
-		} );
-	}
+		const editedComments = keyBy( map( commentIds, commentId => {
+			const comment = this.state.comments[ commentId ];
+			let newLikeValue, newStatusValue;
+
+			if ( isNil( i_like ) ) {
+				// If the comment is not approved anymore, also remove the like, otherwise keep its previous value
+				newLikeValue = 'approved' === status ? comment.i_like : false;
+			} else {
+				newLikeValue = i_like;
+			}
 
-	setCommentStatus = ( commentId, status ) => {
-		const comment = this.state.comments[ commentId ];
-
-		// If the comment is not approved anymore, also remove the like, otherwise keep its previous value
-		this.setState( {
-			comments: {
-				...this.state.comments,
-				[ commentId ]: {
-					...comment,
-					i_like: 'approved' === status ? comment.i_like : false,
-					status,
-				}
-			},
-		} );
+			if ( isNil( status ) ) {
+				newStatusValue = i_like ? 'approved' : comment.status;
+			} else {
+				newStatusValue = status;
+			}
+
+			return {
+				...comment,
+				i_like: newLikeValue,
+				status: newStatusValue,
+			};
+		} ), 'ID' );
+
+		this.setState( { comments: {
+			...this.state.comments,
+			...editedComments,
+		} } );
 	}
 
-	toggleCommentLike = commentId => this.setCommentLike(
-		commentId,
-		! get( this.state.comments, [ commentId, 'i_like' ], false )
-	);
-
-	render() {
-		return (
-			<WrappedCommentList
-				{ ...this.props }
-				comments={ values( this.state.comments ) }
-				deleteCommentPermanently={ this.deleteCommentPermanently }
-				setCommentLike={ this.setCommentLike }
-				setCommentStatus={ this.setCommentStatus }
-				toggleCommentLike={ this.toggleCommentLike }
-			/>
-		);
+	filterCommentsByStatus = () => 'all' === this.props.status
+		? filter( this.state.comments, ( { status } ) => ( 'approved' === status || 'unapproved' === status ) )
+		: filter( this.state.comments, ( { status } ) => ( this.props.status === status ) );
+
+	getCommentsFromProps = ( { comments } ) => this.setState( { comments: keyBy( comments, 'ID' ) } );
+
+	setBulkStatus = ( commentIds, status ) => this.doAction( commentIds, { status } );
+
+	setCommentLike = ( commentId, i_like ) => this.doAction( [ commentId ], { i_like } );
+
+	setCommentStatus = ( commentId, status ) => this.doAction( [ commentId ], { status } );
+
+	/**
+	 * Resets the status and the like value of a list of comments to their previous values.
+	 *
+	 * `comments` is the array of comments that will receive the action.
+	 * It can contain complete comment objects, but in fact it only needs their (previous) `ID`, `i_like`, and `status` properties.
+	 *
+	 * @param {Array}  comments
+	 * @param {int}    comments[].ID
+	 * @param {bool}   comments[].i_like
+	 * @param {string} comments[].status
+	 */
+	undoAction = comments => {
```

Comment:
Just an aside, but depending on how we implement undo later, we'll likely not need this later if we end up storing undo actions in memory.